Unable to get provider leakcanary.internal.AppWatcherInstaller

### Description

Hi!

**Scenario:**
I'm building an instant app through app bundle. At the time of launching the instant app I get a runtime exception.
There is a way to disable leak canary when is instant app?

**StackTrace:**
```
java.lang.RuntimeException: Unable to get provider leakcanary.internal.AppWatcherInstaller: kotlin.KotlinNullPointerException
        at android.app.ActivityThread.installProvider(ActivityThread.java:6988)
        at android.app.ActivityThread.installContentProviders(ActivityThread.java:6528)
        at android.app.ActivityThread.handleBindApplication(ActivityThread.java:6445)
        at android.app.ActivityThread.access$1300(ActivityThread.java:219)
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1859)
        at android.os.Handler.dispatchMessage(Handler.java:107)
        at android.os.Looper.loop(Looper.java:214)
        at android.app.ActivityThread.main(ActivityThread.java:7356)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:492)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:930)
     Caused by: kotlin.KotlinNullPointerException
        at leakcanary.internal.InternalLeakCanary.addDynamicShortcut(InternalLeakCanary.kt:124)
        at leakcanary.internal.InternalLeakCanary.invoke(InternalLeakCanary.kt:83)
        at leakcanary.internal.InternalLeakCanary.invoke(InternalLeakCanary.kt:30)
        at leakcanary.internal.InternalAppWatcher.install(InternalAppWatcher.kt:70)
        at leakcanary.internal.AppWatcherInstaller.onCreate(AppWatcherInstaller.kt:19)
        at android.content.ContentProvider.attachInfo(ContentProvider.java:2092)
        at android.content.ContentProvider.attachInfo(ContentProvider.java:2066)
        at android.app.ActivityThread.installProvider(ActivityThread.java:6983)
        at android.app.ActivityThread.installContentProviders(ActivityThread.java:6528) 
        at android.app.ActivityThread.handleBindApplication(ActivityThread.java:6445) 
        at android.app.ActivityThread.access$1300(ActivityThread.java:219) 
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1859) 
        at android.os.Handler.dispatchMessage(Handler.java:107) 
        at android.os.Looper.loop(Looper.java:214) 
        at android.app.ActivityThread.main(ActivityThread.java:7356) 
        at java.lang.reflect.Method.invoke(Native Method) 
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:492) 
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:930) 
```

### Steps to Reproduce

From an instant enabled dynamic feature:

1. Add leak canary dependency to gradle
2. Run instant app feature

**Expected behavior:**
Instant app should run normally without crashing at launch time.

### Version Information

* LeakCanary version: 2.0-beta-3
* Android OS version: 10
* Gradle version: 5.6.4

### Additional Information

As I could investigate, the problem is here https://github.com/square/leakcanary/blob/53fa1902d3eb3aec9d99257a517dc6ad198acc39/leakcanary-android-core/src/main/java/leakcanary/internal/InternalLeakCanary.kt#L124
Looks like in instant app the `ShortcutManager` service is not available.

--- a/leakcanary-android-core/src/main/java/leakcanary/internal/InternalLeakCanary.kt
+++ b/leakcanary-android-core/src/main/java/leakcanary/internal/InternalLeakCanary.kt
@@ -121,6 +121,10 @@ internal object InternalLeakCanary : (Application) -> Unit, OnObjectRetainedList
     if (!application.resources.getBoolean(R.bool.leak_canary_add_dynamic_shortcut)) {
       return
     }
+    if (VERSION.SDK_INT >= VERSION_CODES.O && application.packageManager.isInstantApp) {
+      // Instant Apps don't have access to ShortcutManager
+      return
+    }
 
     val shortcutManager = application.getSystemService(ShortcutManager::class.java)!!
     val dynamicShortcuts = shortcutManager.dynamicShortcuts

